lib/puppet/provider/rvm_gem/gem.rb

```diff
@@ -76,6 +76,15 @@ def self.gemsplit(desc)
     end
   end
 
+  def gemfile_from_puppet_url(source)
+    unless tmp = Puppet::FileServing::Content.indirection.find(source, :environment => resource.catalog.environment_instance, :links => :follow)
+      fail "Could not find gemfile at %s" % source
+    end
+    tmpfile = Tempfile.open(['puppet-rvm_geminstaller','.gem'])
+    tmpfile.write(tmp.content)
+    tmpfile.flush
+    return tmpfile.path.to_s
```

Comment:
I think this leaks a temporary file for every Puppet run.

Reply:
You are correct the file persists. However, you cannot delete the file right there, as it would no longer be available for the installation procedure.

Unfortunately, I no longer have an instance to validate, but I believe the assumption was this would only be execute if the gem is missing and thus create a file once.